yescrypt-sysendian.h

```diff
@@ -0,0 +1,182 @@
+/*-
```

Comment:
This file almost duplicates the contents of `byteorder.h`.  The functions for 16 bit types could be added there and then yescrypt files can be modified to use them.

Reply:
From a quick check in coverage report, it seems the 16 bit conversions are not used anyways.  You should drop this file and replace it usage with `byteorder.h` as it can be done in a really safe way.

Reply:
I'm OK dropping the unused functions (inherited from scrypt, which in turn shared them with code in FreeBSD - which is why). That's a simple enough change for further merges (from scrypt). I'm not OK with more invasive changes.

Reply:
Dropped the unused `sysendian.h` functions in my yescrypt tree.